lib/cocoapods/installer.rb

```diff
@@ -104,6 +104,8 @@ def prepare
         sandbox.prepare
         ensure_plugins_are_installed!
         Migrator.migrate(sandbox)
+        context = HooksContext.generate(sandbox, podfile, lockfile)
```

Comment:
The HooksContext object is very specific to the `:post_install` is this needed? can I just directly post the sandbox, podfile and lockfile?